File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/ImplementationGuide.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class ImplementationGuide : DomainResource, IResource
{
    public string? Description { get; set; }
    public ImplementationGuide_Definition? Definition { get; set; }
    public string? Date { get; set; }
    public string? Publisher { get; set; }
    public required string[] FhirVersion { get; set; }
    public string? License { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public ImplementationGuide_Global[]? Global_ { get; set; }
    public ImplementationGuide_DependsOn[]? DependsOn { get; set; }
    public required string Name { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public string? Copyright { get; set; }
    public bool? Experimental { get; set; }
    public ImplementationGuide_Manifest? Manifest { get; set; }
    public string? Title { get; set; }
    public required string Status { get; set; }
    public required string Url { get; set; }
    public string? Version { get; set; }
    public required string PackageId { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }

    public class ImplementationGuide_Definition_Page_Page : BackboneElement
    {
        public object? Name    
        {
            get
            {
                if (NameUrl is not null)
                {
                    return NameUrl;
                }
        
                if (NameReference is not null)
                {
                    return NameReference;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(string))
                {
                    NameUrl = (string)value;return;
                }
        
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    NameReference = (Base.ResourceReference)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public string[]? Page { get; set; }
        public required string Title { get; set; }
        public string? NameUrl { get; set; }
        public required string Generation { get; set; }
        public Base.ResourceReference? NameReference { get; set; }
    }

    public class ImplementationGuide_Definition_Page : BackboneElement
    {
        public object? Name    
        {
            get
            {
                if (NameUrl is not null)
                {
                    return NameUrl;
                }
        
                if (NameReference is not null)
                {
                    return NameReference;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(string))
                {
                    NameUrl = (string)value;return;
                }
        
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    NameReference = (Base.ResourceReference)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public ImplementationGuide_Definition_Page_Page[]? Page { get; set; }
        public required string Title { get; set; }
        public string? NameUrl { get; set; }
        public required string Generation { get; set; }
        public Base.ResourceReference? NameReference { get; set; }
    }

    public class ImplementationGuide_Definition_Grouping : BackboneElement
    {
        public required string Name { get; set; }
        public string? Description { get; set; }
    }

    public class ImplementationGuide_Definition_Resource : BackboneElement
    {
        public string? Name { get; set; }
        public object? Example    
        {
            get
            {
                if (ExampleBoolean is not null)
                {
                    return ExampleBoolean;
                }
        
                if (ExampleCanonical is not null)
                {
                    return ExampleCanonical;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(bool))
                {
                    ExampleBoolean = (bool)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ExampleCanonical = (string)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public required Base.ResourceReference Reference { get; set; }
        public string? GroupingId { get; set; }
        public string? Description { get; set; }
        public string[]? FhirVersion { get; set; }
        public bool? ExampleBoolean { get; set; }
        public string? ExampleCanonical { get; set; }
    }

    public class ImplementationGuide_Definition_Template : BackboneElement
    {
        public required string Code { get; set; }
        public string? Scope { get; set; }
        public required string Source { get; set; }
    }

    public class ImplementationGuide_Definition_Parameter : BackboneElement
    {
        public required string Code { get; set; }
        public required string Value { get; set; }
    }

    public class ImplementationGuide_Definition : BackboneElement
    {
        public ImplementationGuide_Definition_Page? Page { get; set; }
        public ImplementationGuide_Definition_Grouping[]? Grouping { get; set; }
        public required ImplementationGuide_Definition_Resource[] Resource { get; set; }
        public ImplementationGuide_Definition_Template[]? Template { get; set; }
        public ImplementationGuide_Definition_Parameter[]? Parameter { get; set; }
    }

    public class ImplementationGuide_Global : BackboneElement
    {
        public required string Type { get; set; }
        public required string Profile { get; set; }
    }

    public class ImplementationGuide_DependsOn : BackboneElement
    {
        public required string Uri { get; set; }
        public string? Version { get; set; }
        public string? PackageId { get; set; }
    }

    public class ImplementationGuide_Manifest_Page : BackboneElement
    {
        public required string Name { get; set; }
        public string? Title { get; set; }
        public string[]? Anchor { get; set; }
    }

    public class ImplementationGuide_Manifest_Resource : BackboneElement
    {
        public object? Example    
        {
            get
            {
                if (ExampleBoolean is not null)
                {
                    return ExampleBoolean;
                }
        
                if (ExampleCanonical is not null)
                {
                    return ExampleCanonical;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(bool))
                {
                    ExampleBoolean = (bool)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ExampleCanonical = (string)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public required Base.ResourceReference Reference { get; set; }
        public string? RelativePath { get; set; }
        public bool? ExampleBoolean { get; set; }
        public string? ExampleCanonical { get; set; }
    }

    public class ImplementationGuide_Manifest : BackboneElement
    {
        public ImplementationGuide_Manifest_Page[]? Page { get; set; }
        public string[]? Image { get; set; }
        public string[]? Other { get; set; }
        public required ImplementationGuide_Manifest_Resource[] Resource { get; set; }
        public string? Rendering { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/ImplementationGuide.py
from typing import Optional, List
from pydantic import *
from base.UsageContext import UsageContext
from base.ContactDetail import ContactDetail
from base.CodeableConcept import CodeableConcept
from base.DomainResource import DomainResource
from base.Reference import Reference
from base.BackboneElement import BackboneElement

class ImplementationGuide_Definition_Page(BackboneElement):
    page: Optional[List[Reference]] = None
    title: str
    name_url: Optional[str] = None
    generation: str
    name_reference: Optional[Reference] = None

class ImplementationGuide_Definition_Grouping(BackboneElement):
    name: str
    description: Optional[str] = None

class ImplementationGuide_Definition_Resource(BackboneElement):
    name: Optional[str] = None
    reference: Reference
    grouping_id: Optional[str] = None
    description: Optional[str] = None
    fhir_version: Optional[List[str]] = None
    example_boolean: Optional[bool] = None
    example_canonical: Optional[str] = None

class ImplementationGuide_Definition_Template(BackboneElement):
    code: str
    scope: Optional[str] = None
    source: str

class ImplementationGuide_Definition_Parameter(BackboneElement):
    code: str
    value: str

class ImplementationGuide_Definition(BackboneElement):
    page: Optional[ImplementationGuide_Definition_Page] = None
    grouping: Optional[List[ImplementationGuide_Definition_Grouping]] = None
    resource: list[ImplementationGuide_Definition_Resource] = []
    template: Optional[List[ImplementationGuide_Definition_Template]] = None
    parameter: Optional[List[ImplementationGuide_Definition_Parameter]] = None

class ImplementationGuide_Global(BackboneElement):
    type: str
    profile: str

class ImplementationGuide_DependsOn(BackboneElement):
    uri: str
    version: Optional[str] = None
    package_id: Optional[str] = None

class ImplementationGuide_Manifest_Page(BackboneElement):
    name: str
    title: Optional[str] = None
    anchor: Optional[List[str]] = None

class ImplementationGuide_Manifest_Resource(BackboneElement):
    reference: Reference
    relative_path: Optional[str] = None
    example_boolean: Optional[bool] = None
    example_canonical: Optional[str] = None

class ImplementationGuide_Manifest(BackboneElement):
    page: Optional[List[ImplementationGuide_Manifest_Page]] = None
    image: Optional[List[str]] = None
    other: Optional[List[str]] = None
    resource: list[ImplementationGuide_Manifest_Resource] = []
    rendering: Optional[str] = None

class ImplementationGuide(DomainResource):
    description: Optional[str] = None
    definition: Optional[ImplementationGuide_Definition] = None
    date: Optional[str] = None
    publisher: Optional[str] = None
    fhir_version: list[str] = []
    license: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = None
    global: Optional[List[ImplementationGuide_Global_]] = None
    depends_on: Optional[List[ImplementationGuide_DependsOn]] = None
    name: str
    use_context: Optional[List[UsageContext]] = None
    copyright: Optional[str] = None
    experimental: Optional[bool] = None
    manifest: Optional[ImplementationGuide_Manifest] = None
    title: Optional[str] = None
    status: str
    url: str
    version: Optional[str] = None
    package_id: str
    contact: Optional[List[ContactDetail]] = None